sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/BaseScriptTransformationSuite.scala

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.execution
+
+import java.sql.Timestamp
+import java.util.Locale
+
+import org.scalatest.Assertions._
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.exceptions.TestFailedException
+
+import org.apache.spark.{SparkException, TaskContext, TestUtils}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Column
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference, Expression}
+import org.apache.spark.sql.catalyst.plans.physical.Partitioning
+import org.apache.spark.sql.execution._
+import org.apache.spark.sql.hive.HiveUtils
+import org.apache.spark.sql.hive.test.TestHiveSingleton
+import org.apache.spark.sql.test.SQLTestUtils
+import org.apache.spark.sql.types.StringType
+
+abstract class BaseScriptTransformationSuite extends SparkPlanTest with SQLTestUtils
+  with TestHiveSingleton with BeforeAndAfterEach {
+
+  def scriptType: String
+
+  import spark.implicits._
+
+  var noSerdeIOSchema: BaseScriptTransformIOSchema = _
+
+  private var defaultUncaughtExceptionHandler: Thread.UncaughtExceptionHandler = _
+
+  protected val uncaughtExceptionHandler = new TestUncaughtExceptionHandler
+
+  protected override def beforeAll(): Unit = {
+    super.beforeAll()
+    defaultUncaughtExceptionHandler = Thread.getDefaultUncaughtExceptionHandler
+    Thread.setDefaultUncaughtExceptionHandler(uncaughtExceptionHandler)
+  }
+
+  protected override def afterAll(): Unit = {
+    super.afterAll()
+    Thread.setDefaultUncaughtExceptionHandler(defaultUncaughtExceptionHandler)
+  }
+
+  override protected def afterEach(): Unit = {
+    super.afterEach()
+    uncaughtExceptionHandler.cleanStatus()
+  }
+
+  def createScriptTransformationExec(
+      input: Seq[Expression],
+      script: String,
+      output: Seq[Attribute],
+      child: SparkPlan,
+      ioschema: BaseScriptTransformIOSchema): BaseScriptTransformationExec = {
+    scriptType.toUpperCase(Locale.ROOT) match {
+      case "SPARK" => new SparkScriptTransformationExec(
```

Comment:
instead of asking the child to override `scriptType`, we should just ask child to implement a method to create `BaseScriptTransformationExec`.

Then we can move the base suite to sql/core.

Reply:
> instead of asking the child to override `scriptType`, we should just ask child to implement a method to create `BaseScriptTransformationExec`.
> 
> Then we can move the base suite to sql/core.

There are inherit confit between `SparkPlanTest` and ` SQLTestUtils`

Reply:
cc @cloud-fan  when move BaseScriptTransformationSuite to sql/core there are method conflict like below, how can I solve this is better?
```
[error] /Users/angerszhu/Documents/project/AngersZhu/spark/sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationSuite.scala:31: overriding method spark in trait SharedSparkSessionBase of type => org.apache.spark.sql.SparkSession;
[error]  value spark in trait TestHiveSingleton of type org.apache.spark.sql.SparkSession has weaker access privileges; it should be public
[error] class HiveScriptTransformationSuite extends BaseScriptTransformationSuite with TestHiveSingleton {
[error]       ^
[error] on
```